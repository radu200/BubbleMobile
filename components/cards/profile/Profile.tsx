import React, { useMemo } from "react";
import { StyleSheet, Text, View, Image } from "react-native";
import Colors from "../../../constants/Colors";
import StaticData from '../../../data/static'
import DefaultAvatar from '../../../assets/images/profile_default.png';
import ProfileImageExmaple from "../../../assets/images/profile-image-example.png";
import Children from './Children';
import { profileTypes } from '../../../types'

interface props {
  user: profileTypes,
}

export default function Profile({ user }: props) {
  const {
    address,
    aboutMe,
    contacts
  } = StaticData.profile
  return (
    <View style={styles.card}>
      <View style={styles.header}>
        <Image
          style={styles.avatar}
          source={user.profileImageUrl ? ProfileImageExmaple : DefaultAvatar}
        />
        <Text style={styles.userName}>
          {user.fullName}
        </Text>
        <Text style={styles.accountTotal}>{user.accountTotal} {StaticData.currency.GBP} </Text>
      </View>
      <View style={styles.container}>
        <Text style={styles.title}>{contacts}</Text>
        <Text>
          {user.mobileNumber}
        </Text>
        <Text>
          {user.email}
        </Text>
      </View>
      <View style={styles.container}>
        <Text style={styles.title}>{aboutMe}</Text>
        <Text>
          {user.biography}
        </Text>
      </View>

      <View style={styles.container}>
        <Children data={user.children} />
      </View>

      <View style={styles.container}>
        <Text style={styles.title}>{address.title}</Text>
        <Text>
          {address.houseNo}: {user.address.houseNo}
        </Text>
        <Text>
          {address.postcode}: {user.address.postcode}
        </Text>
        <Text>
          {address.street}: {user.address.street}
        </Text>
        <Text>
          {address.town}:  {user.address.town}
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  card: {
    borderRadius: 10,
    backgroundColor: Colors.light.background,
    marginBottom: 15,
    padding: 20,
    shadowColor: Colors.dark.background,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  header: {
    alignItems: 'center'
  },
  container: {
    paddingTop: 10,
    paddingBottom: 10,
    borderBottomColor: Colors.light.tint,
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  avatar: {
    width: 120,
    height: 120,
    borderRadius: 75,
  },

  userName: {
    fontSize: 20,
    fontWeight: "bold",
    paddingTop: 10,
    paddingBottom: 8,
  },
  accountTotal: {
    fontSize: 16,
    fontWeight: "bold",
    color: Colors.light.tint
  },

  title: {
    fontSize: 16,
    fontWeight: "bold",
    paddingBottom: 5,
  }
});
