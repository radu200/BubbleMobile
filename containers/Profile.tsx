import React, { useEffect } from "react";
import { connect, ConnectedProps } from "react-redux";
import { fetchUser } from '../redux/profile/operators';
import { profileTypes } from '../types';
import { RootState } from '../redux/rootReducer'
import ProfileScreen from "../screens/Profile";


interface props extends PropsFromRedux {
  fetchUser: any,
  user: profileTypes | any,
  spinner: boolean,
  failure: boolean
}

const Profile = ({ fetchUser, user, failure, spinner }: props) => {
  useEffect(() => {
    fetchUser();
  }, []);

  return <ProfileScreen user={user} spinner={spinner} failure={failure} />
}

const mapState = (state: RootState) => ({
  user: state.profile.user,
  spinner: state.profile.spinner,
  failure: state.profile.failure
})

const connector = connect(
  mapState,
  { fetchUser }
)

type PropsFromRedux = ConnectedProps<typeof connector>

export default connector(Profile);