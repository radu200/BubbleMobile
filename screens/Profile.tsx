import React from "react";
import { StyleSheet, View } from "react-native";
import Spinner from '../components/progress/Spinner';
import ErrorAlert from '../components/alerts/Error';
import StaticData from '../data/static';
import ProfileCard from '../components/cards/profile/Profile';
import { profileTypes } from '../types'

interface props {
  user: profileTypes,
  spinner: boolean,
  failure: boolean,
}

export default function ProfileScreen({
  user,
  spinner,
  failure,
}: props) {

  return (
    <>
      {failure ? <ErrorAlert msg={StaticData.msg.error} /> : null}
      {spinner ? <Spinner /> :
        <View style={styles.container}>
          {user && <ProfileCard user={user} />}
        </View>}
    </>

  );
}

const styles = StyleSheet.create({
  container: {
    margin: 15,
  },
});
