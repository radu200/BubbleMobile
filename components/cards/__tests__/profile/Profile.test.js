import React from 'react';
import renderer from 'react-test-renderer';
import Profile from '../../profile/Profile';

it('Profile component renders correctly', () => {
    const user = {
        accountTotal: 0,
        profileImageUrl: "url",
        fullName: "Janet Stevans",
        email: "email@j.com",
        mobileNumber: "76236828",
        biography: "about me",
        address: {
            houseNo: "2",
            postcode: "3BD ST",
            street: "James Bond",
            town: "London"
        },
        children: [{
            age: 2,
            gender: 0
        }]
    }

    const tree = renderer
        .create(<Profile user={user} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})

