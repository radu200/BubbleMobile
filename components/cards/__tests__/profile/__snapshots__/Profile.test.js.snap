// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "elevation": 5,
      "marginBottom": 15,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 75,
          "height": 120,
          "width": 120,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingBottom": 8,
          "paddingTop": 10,
        }
      }
    >
      Janet Stevans
    </Text>
    <Text
      style={
        Object {
          "color": "#55427A",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      0
       
      GBP
       
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#55427A",
        "borderBottomWidth": 0.5,
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 5,
        }
      }
    >
      Contacts
    </Text>
    <Text>
      76236828
    </Text>
    <Text>
      email@j.com
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#55427A",
        "borderBottomWidth": 0.5,
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 5,
        }
      }
    >
      About me
    </Text>
    <Text>
      about me
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#55427A",
        "borderBottomWidth": 0.5,
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 5,
        }
      }
    >
      Children
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "margin": 5,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "margin": 5,
              "width": 25,
            }
          }
        />
        <Text>
          2
           
          years
           
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#55427A",
        "borderBottomWidth": 0.5,
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 5,
        }
      }
    >
      Address
    </Text>
    <Text>
      House Number
      : 
      2
    </Text>
    <Text>
      Postcode
      : 
      3BD ST
    </Text>
    <Text>
      Street
      : 
      James Bond
    </Text>
    <Text>
      Town
      :  
      London
    </Text>
  </View>
</View>
`;
